src/block_service/hdfs/hdfs_service.cpp

```diff
@@ -55,6 +56,12 @@ DSN_DEFINE_uint64("replication",
                   "hdfs write batch size, the default value is 64MB");
 DSN_TAG_VARIABLE(hdfs_write_batch_size_bytes, FT_MUTABLE);
 
+DSN_DEFINE_bool("replication",
+                hdfs_enable_direct_io,
+                false,
+                "Whether to enable direct I/O when download files from hdfs");
+DSN_TAG_VARIABLE(hdfs_enable_direct_io, FT_MUTABLE);
```

Comment:
Actually, it's not `hdfs_direct_io`, you just only support saving hdfs file to local via direct_io.

You may consider that define general `enable_direct_io`  in `block_service.h` or `directio_writable_file.h`  and used for any remote filesystem to save file